core/src/com/unciv/ui/EmpireOverviewScreen.kt

```diff
@@ -358,8 +374,16 @@ class EmpireOverviewScreen(val viewingPlayer:CivilizationInfo) : CameraStageBase
                 val statusLine = ImageGetter.getLine(civGroup.x+civGroup.width/2,civGroup.y+civGroup.height/2,
                         otherCivGroup.x+otherCivGroup.width/2,otherCivGroup.y+otherCivGroup.height/2,3f)
 
-                statusLine.color = if(diplomacy.diplomaticStatus== DiplomaticStatus.War) Color.RED
-                else Color.GREEN
+                if(diplomacy.diplomaticStatus == DiplomaticStatus.War)
+                    statusLine.color = Color.RED
+                else {
+                    val diplomacyLevel = diplomacy.diplomaticModifiers.values.sum()
+                    statusLine.color = getColorForDiplomacyLevel(diplomacyLevel)
+                }
+
+                if (civLines[civ.civName] == null)
```

Comment:
Please change to civLines.containsKey, this looks like we set the value to null somewhere? Do we?

Reply:
First of all, thank you for really good review =)
I was assuming that comparing with `null` is more cheap operation rather than calling a method to check the key. Anyway, if it decreases readability of code, I will correct that since the gain is not big.